src/rp2_common/pico_tone/tone.c

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2020 Raspberry Pi (Trading) Ltd.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+#include "pico/stdlib.h"
+#include "pico/tone.h"
+#include "hardware/pwm.h"
+
+void tone_init(uint gpio) {
+    // Configure GPIO for PWM output
+    gpio_set_function(gpio, GPIO_FUNC_PWM);
+    // Find out which PWM slice is connected to GPIO
+    uint slice_num = pwm_gpio_to_slice_num(gpio);
+    // Get default configuration for PWM slice and initialise PWM with it
+    pwm_config config = pwm_get_default_config();
+    pwm_init(slice_num, &config, true);
+}
+
+void no_tone(uint gpio) {
+    pwm_set_gpio_level(gpio, 0);
+}
+
+void tone(uint gpio, uint freq, uint32_t duration_ms) {
+    // Calculate and cofigure new clock divider according to the frequency
+    // This formula is assuming we are running at 125MHz.
+    // TODO: Make this work for any frequency
+    float clkdiv = (1.f / freq) * 2000.f;
+    uint slice_num = pwm_gpio_to_slice_num(gpio);
+    pwm_set_clkdiv(slice_num, clkdiv);
+    // Configure duty to 50% ((2**16)-1)/2) to generate a square wave
+    pwm_set_gpio_level(gpio, 32768U);
+    sleep_ms(duration_ms);
+
+    // Make silence for some ms to distinguish between tones
+    no_tone(gpio);
+    sleep_ms(PICO_TONE_SILENCE_DELAY_MS);
```

Comment:
Probably makes more sense to pass this in as a function-parameter, rather than forcing every `tone` to always have the **same** amount of silence after it?

Reply:
I was actually thinking more about this and I think it is better not to have any silence at all and keep the this up to the user as there might be applications where this silence is not needed.

I have make the required changes to the `tone.c` and `tone.h` files.